File: Framework/Domain/ForeignKeyMapping.cs
using System.Collections.Generic;

namespace Framework.Domain
{
    public interface IForeignKeyMapping<TParentEntity, TChildEntity>
    {
        IList<TChildEntity> GetChildEntities(TParentEntity parent);
        IList<TChildEntity> RetrieveForeignKeyEntities(TParentEntity parent);
    }

    public abstract class ForeignKeyMapping<TParentEntity, TChildEntity> : IForeignKeyMapping<TParentEntity, TChildEntity>
    {
        public IList<TChildEntity> GetChildEntities(TParentEntity parent)
        {
            return RetrieveForeignKeyEntities(parent);
        }

        public abstract IList<TChildEntity> RetrieveForeignKeyEntities(TParentEntity parent);
    }
}

File: Android/app/src/main/java/com/peaa/Domain/ForeignKeyMapping.java
package com.peaa.Domain;

import com.peaa.Domain.Interfaces.IForeignKeyMapping;

import java.util.List;

/**
 * Created by aiko on 3/4/17.
 */

public abstract class ForeignKeyMapping<TParentEntity, TChildEntity>
        implements IForeignKeyMapping<TParentEntity, TChildEntity>
{
    public List<TChildEntity> GetChildEntities(TParentEntity parent)
    {
        return RetrieveForeignKeyEntities(parent);
    }

    public abstract List<TChildEntity> RetrieveForeignKeyEntities(TParentEntity parent);
}

